src/git/remote.rs

```diff
@@ -48,3 +65,23 @@ pub async fn git_remote_set_url(
 
     Ok(())
 }
+
+pub async fn git_remote_update_repo_name(
+    repo: &path::Path,
+    new_name: &str,
+) -> Result<(), errors::Error> {
+    info!("Updating git remote URL to match the new name");
+
+    // git remote get-url origin
+
+    let remote = git_remote_get_url(repo, "origin").await?
+        .into_iter()
+        .next();
+
+    if let Some(remote_url) = remote.and_then(|url| GitRemote::parse(&url)) {
```

Comment:
If I'm reading the associated logic right, the goal here is to parse the configured `origin` URL, replace just the repo name, and then set that right?

I suspect we can probably make the assumption that if the repo is being managed using Git-Tool already, the URL will have been set using the `tasks::GitRemote` task and so we can just use that (with a `Repo` corresponding to the post-rename variant) to update to the correct URL.

That has the added benefit of saving you needing to parse the various different URI formats supported by Git or getting caught by the infuriating mess that is non-standard Git servers with random usernames and ports 😄.